### PR TITLE
clippy: arithmetic-side-effects

### PR DESCRIPTION
**Pull request by yihau**
(part of #2487)

#### Problem

Screenshot 2024-08-08 at 21 54 33

#### Summary of Changes

replace `add_assign` with `saturating_add`

### CODE REVIEW

File: sdk/program/src/epoch_rewards.rs [hunk 1]
```diff
@@ -44,7 +44,7 @@ impl EpochRewards {
     pub fn distribute(&mut self, amount: u64) {
         assert!(self.distributed_rewards.saturating_add(amount) <= self.total_rewards);
 
-        self.distributed_rewards.add_assign(amount);
+        self.distributed_rewards = self.distributed_rewards.saturating_add(amount);
```
Comment (CriesofCarrots) [inline sdk/program/src/epoch_rewards.rs]: The lint seems unnecessary here, but the change to saturating add is innocuous. I don't think we need to do the addition multiple times, though
How about this:
```
pub fn distribute(&mut self, amount: u64) {
    let new_distributed_rewards = self.distributed_rewards.saturating_add(amount);
    assert!(new_distributed_rewards <= self.total_rewards);
    self.distributed_rewards = new_distributed_rewards;
}
```

Comment (HaoranYi) [reply sdk/program/src/epoch_rewards.rs]: I like it.

Comment (yihau) [reply sdk/program/src/epoch_rewards.rs]: thank you for the insight! yeah, good catch! just fixed it! [34f81e7](https://github.com/anza-xyz/agave/pull/2493/commits/34f81e7e653d33e64ae1d220721936dfac980137)

**Review by CriesofCarrots — DISMISSED** [review]

**Comment by CriesofCarrots** [comment]
🤔 I'll look at that test failure.

**Comment by yihau** [comment]
ah... sorry for missing the update for the test. just fixed it!

**Review by CriesofCarrots — DISMISSED** [review]
Oh nice, thanks!

**Review by CriesofCarrots — APPROVED** [review]

**Pull request merged**